pkg/report/datafeeds/processor/processor.go

```diff
@@ -0,0 +1,94 @@
+package processor
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder"
+	"github.com/smartcontractkit/chainlink-evm/pkg/report/datafeeds"
+
+	df "github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/monitoring/pb/data-feeds/on-chain/registry"
+	wt "github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/monitoring/pb/platform"
+)
+
+func NewDataFeedsProcessor(metrics *df.Metrics, emitter beholder.ProtoEmitter) *datafeeds.Processor {
+	return datafeeds.NewProcessor(metrics, emitter,
+		GetDataFeedsSchema(),
+		Decode,
+	)
+}
+
+func NewCCIPDataFeedsProcessor(metrics *df.Metrics, emitter beholder.ProtoEmitter) *datafeeds.Processor {
+	return datafeeds.NewProcessor(metrics, emitter,
+		GetCCIPDataFeedsSchema(),
+		DecodeCCIP,
+	)
+}
+
+func GetDataFeedsSchema() abi.Arguments {
+	return datafeeds.GetSchema("tuple(bytes32,uint32,uint224)[]", "", []abi.ArgumentMarshaling{
+		{Name: "feedID", Type: "bytes32"},
+		{Name: "timestamp", Type: "uint32"},
+		{Name: "price", Type: "uint224"},
+	})
+}
+
+func GetCCIPDataFeedsSchema() abi.Arguments {
+	return datafeeds.GetSchema("tuple(bytes32,uint224,uint32)[]", "", []abi.ArgumentMarshaling{
+		{Name: "feedID", Type: "bytes32"},
+		{Name: "price", Type: "uint224"},
+		{Name: "timestamp", Type: "uint32"},
+	})
+}
+
+// Decode is made available to external users (i.e. mercury server)
+func Decode(m *wt.WriteConfirmed, data []byte, schema abi.Arguments) ([]*df.FeedUpdated, error) {
```

Comment:
Sorry, something is off here... This is underlying report decoding - it has nothing to do with our WT, with our messages, monitoring, anything on the higher levels. This is just low-level binding to a specific type and decoding in/out that type.

Please decouple this implementation from using outer types (e.g., `wt.WriteConfirmed`) and have the outer layer (WT, WT monitor) use the decoder.

Reply:
followup ticket: https://smartcontract-it.atlassian.net/browse/PLEX-1535